File: system-network/src/main/java/com/levnovikov/core_network/response/Response.java
package com.levnovikov.core_network.response;

/**
 * Author: lev.novikov
 * Date: 27/3/18.
 */

public class Response {

    private int code;
    private ResponseBody body;

    private Response() {

    }

    public int getCode() {
        return code;
    }

    public ResponseBody getBody() {
        return body;
    }

    public static class Builder {
        private final Response response = new Response();

        public Builder code(int code) {
            response.code = code;
            return this;
        }

        public Builder body(ResponseBody body) {
            response.body = body;
            return this;
        }

        public Response build() {
            return response;
        }
    }
}

File: system-http-client/src/main/java/com/levnovikov/core_network/response/Response.kt
package com.levnovikov.core_network.response

/**
 * Author: lev.novikov
 * Date: 27/3/18.
 */

class Response private constructor() {

    var code: Int = 0
        private set
    lateinit var body: ResponseBody

    class Builder {
        private val response = Response()

        fun code(code: Int): Builder {
            response.code = code
            return this
        }

        fun body(body: ResponseBody): Builder {
            response.body = body
            return this
        }

        fun build(): Response {
            return response
        }
    }
}
